bindings/rust/s2n-tls-tokio/Cargo.toml

```diff
@@ -15,9 +15,13 @@ default = []
 errno = { version = "0.3" }
 libc = { version = "0.2" }
 pin-project-lite = { version = "0.2" }
-s2n-tls = { version = "=0.1.2", path = "../s2n-tls" }
+s2n-tls = { version = "=0.2.0", path = "../s2n-tls" }
 tokio = { version = "1", features = ["net", "time"] }
 
+[build-dependencies]
+home = "=0.5.5" # a transitive dependency of aws-lc-sys. newer versions require rust 1.70
+regex = "=1.9.6" # a transitive dependency of aws-lc-sys. newer versions require rust 1.65, see https://github.com/aws/s2n-tls/issues/4242
```

Comment:
I started investigating this on other platforms and I dont see an issue when building this on linux. Converting this to draft while I finish investigating

Reply:
aws-lc-rs was behaving differently on Mac vs Linux because of bindgen build logic on those platforms. aws-lc has a [PR](https://github.com/aws/aws-lc-rs/pull/331) to to pin bindgen deps and test it in CI.

---

The behavior I was seeing:
Mac:
- new git clone: fails ‘cargo build’
- new git clone: fails ‘cargo publish --dry-run’

Linux:
- new git clone: success ‘cargo build’
- new git clone: success ‘cargo publish --dry-run’

Reply:
> This doesn't feel like the right approach to me. Pinning transitive(!) dependencies like this is forcing the costs of lower MSRV on all downstream consumers

Its not ideal but in order to guarantee that the project can be built and run with the MSRV we advertise, we need to restrict these. I think that the pain we live with until we can update to a higher MSRV. Hopefully that day is not too far in the future.

Reply:
This isn't the right approach. It needs to be done in `dev-dependencies`. Applications will, unfortunately, need to do the same if they want to build on older rustc versions.

Reply:
Spoke offline, seems that even though build-deps is what these dependencies are being used for, we want to pin them at the dev-deps since those dont get passed onto the application.

Reply:
updated in https://github.com/aws/s2n-tls/pull/4388/commits/1e14a1491d1fe457487f510336baa9f029f7831d

Reply:
> This isn't the right approach. It needs to be done in `dev-dependencies`. Applications will, unfortunately, need to do the same if they want to build on older rustc versions.

I see, so the goal of this is purely for your own CI to check that you're still satisfying msrv? I think typically folks do that by checking in a minimum version compatible lockfile rather than needing to manually do things like this.

Reply:
> checking in a minimum version compatible lockfile
What is this method? Do you have a link to a sample proj or link to a tool?